ERCS/erc-7007.md

```diff
@@ -133,14 +182,40 @@ ERC-7007 Metadata JSON Schema for reference
         "aigc_data": {
             "type": "string",
             "description": "A URI pointing to a resource with mime type image/* representing the asset to which this AIGC NFT represents."
+        },
+        "proof_type": {
+            "type": "string",
+            "description": "validity (zkML) or fraud (opML)"
         }
     }
 }
 ```
 
 ## Rationale
 
-TBD
+### Unique Token Identification
+
+This specification sets the `tokenId` to be the hash of its corresponding `prompt`, creating a deterministic and collision-resistant way to associate tokens with their unique content generation parameters. This design decision ensures that the same prompt (which corresponds to the same AI-generated content under the same model seed) cannot be minted more than once, thereby preventing duplication and preserving the uniqueness of each NFT within the ecosystem.
+
+### Generalization to Different Proof Types
+
+This specification accommodates two proof types: validity proofs for zkML and fraud proofs for opML. Function arguments in `mint` and `verify` are designed for generality, allowing for compatibility with both proof systems. Moreover, the specification includes an updatable extension that specifically serves the requirements of opML.
+
+### `verify` interface
+
+We specify a `verify` interface to enforce the correctness of `aigcData`. It is defined as a view function to reduce gas cost. `verify` SHOULD return true if and only if `aigcData` is finalized in both zkML and opML. In zkML, it MUST verify the ZK proof, i.e. `proof`; in opML, it MUST make sure that the challenging period is finalized, and that the `aigcData` is up-to-date, i.e. has been updated after finalization. Additionally, `proof` CAN be *empty* in opML.
```

Comment:
Avoid using the uppercase keywords ("SHOULD") outside the Specification section, because they create requirements on implementations. If you're explaining something, just use "should" instead.